### DISCUSSION TITLE
idea: consider adding gitops architecture terminology to glossary

### DISCUSSION THREAD

**Discussion by johndietz** [open-gitops/project#139] [Ideas]
there is a common architecture decision that needs to be made when you may have a management gitops cluster/system in play. the decision is whether that management cluster will interact with downstream clusters directly (for this discussion starter: `gitops hub and spoke pattern`) or if workload clusters will bootstrap themselves with their own argocd (to start somewhere, `gitops bootstrap pattern`)

the scenario applies at least to to argocd, but thought conceptually it may apply to every gitops ecosystem - i don't have the expertise to know if that's true. if it is, i would like to propose we define am open term for these two patterns that to soften a complex topic with consistency.

proposal scenario: a management cluster, a preprod cluster, and a production cluster, and commitment to leveraging gitops on [argocd will be my example]

architecture decision:
`gitops hub and spoke pattern`: argocd runs in the management cluster, and manages all apps in management, preprod, and production. there is no argocd needed in the preprod and production clusters.
`gitops bootstrap pattern`: argocd runs in each of management, preprod, and production. each instance of argocd only manages apps in its respective cluster, allowing production isolation from management.

it may be (1) too opinionated (2) too broad (3) too specific to argocd's architecture. they're common enough in my discussions that i thought i'd bring it to the group and collect more thoughts - it would be nice to use terms we all could lean on consistently if these 2 architectures are prevalent throughout the gitops ecosystem.

i'll hop in and join the group tomorrow to speak to it some.

**Comment by todaywasawesome** [comment]
@johndietz I just put together a comprehensive overview of architectures for Argo CD in this blog https://codefresh.io/blog/a-comprehensive-overview-of-argo-cd-architectures-2023/

What blew my mind is that it's actually expanding beyond just these two, there are more now!
